src/Share/Postgres.hs

```diff
@@ -128,6 +131,14 @@ newtype Pipeline e a = Pipeline {unPipeline :: Hasql.Pipeline.Pipeline (Either (
 pipelined :: Pipeline e a -> Transaction e a
 pipelined p = Transaction (Hasql.pipeline (unPipeline p))
 
+-- | Like fmap, but the provided function can throw a recoverable error by returning 'Left'.
+pEitherMap :: (a -> Either e b) -> Pipeline e a -> Pipeline e b
```

Comment:
Ah I see, we don't expose `e` in QueryA, if you have ideas on re-organizing these classes let me know :) 

I'm wondering if maybe having the `e` as part of the Transaction type is a bad idea and maybe we should just have a `tryTransaction :: ExceptT e Transaction a -> m (Either e a)` which knows to roll-back instead; then I guess we'd just use `Compose (Either e) Pipeline` for applicative situations?

At any rate that's a conversation for outside of this PR :)